libnavigator/src/main/java/com/mapbox/navigation/navigator/MapboxNativeNavigatorImpl.kt

```diff
@@ -96,65 +137,188 @@ object MapboxNativeNavigatorImpl : MapboxNativeNavigator {
+     */
     override fun getElectronicHorizon(request: String): RouterResult =
         navigator.getElectronicHorizon(request)
 
     // Offline
 
+    /**
+     * Caches tiles around last set route
+     */
     override fun cacheLastRoute() {
         navigator.cacheLastRoute()
     }
 
+    /**
+     * Configures routers for getting routes offline.
+     *
+     * @param routerParams Optional [RouterParams] object which contains router configurations for
+     * getting routes offline.
+     * @param httpClient A platform specific [HttpInterface]. Can be null so default
+     * implementation will be used.
+     *
+     * @return number of tiles founded in the directory
+     */
     override fun configureRouter(routerParams: RouterParams, httpClient: HttpInterface?): Long =
         navigator.configureRouter(routerParams, httpClient)
 
+    /**
+     * Uses valhalla and local tile data to generate mapbox-directions-api-like json.
+     *
+     * @param url the directions-based uri used when hitting the http service
+     * @return a [RouterResult] object with the json and a success/fail bool
+     */
     override fun getRoute(url: String): RouterResult = navigator.getRoute(url)
 
+    /**
+     * Passes in an input path to the tar file and output path.
+     *
+     * @param tarPath The path to the packed tiles.
+     * @param destinationPath The path to the unpacked files.
+     *
+     * @return the number of unpacked tiles
+     */
     override fun unpackTiles(tarPath: String, destinationPath: String): Long =
         navigator.unpackTiles(tarPath, destinationPath)
 
+    /**
+     * Removes tiles wholly within the supplied bounding box. If the tile is not
+     * contained completely within the bounding box it will remain in the cache.
+     * After removing files from the cache any routers should be reconfigured
+     * to synchronize their in memory cache with the disk.
+     *
+     * @param tilePath The path to the tiles.
+     * @param southwest The lower left coord of the bbox.
+     * @param northeast The upper right coord of the bbox.
+     *
+     * @return the number of tiles removed
+     */
     override fun removeTiles(tilePath: String, southwest: Point, northeast: Point): Long =
         navigator.removeTiles(tilePath, southwest, northeast)
 
     // History traces
-
+    /**
+     * Gets the history of state changing calls to the navigator this can be used to
+     * replay a sequence of events for the purpose of bug fixing.
+     *
+     * @return a json representing the series of events that happened since the last time
+     * history was toggled on
+     */
     override fun getHistory(): String = navigator.history
 
+    /**
+     * Toggles the recording of history on or off.
+     * Toggling will reset all history call [getHistory] first before toggling to retain a copy.
+     *
+     * @param isEnabled set this to true to turn on history recording and false to turn it off
+     */
```

Comment:
```suggestion
    /**
     * Gets the history of state-changing calls to the navigator. This can be used to
     * replay a sequence of events for the purpose of bug fixing.
     *
     * @return a json representing the series of events that happened since the last time
     * the history was toggled on.
     */
    override fun getHistory(): String = navigator.history

    /**
     * Toggles the recording of history on or off.
     * Toggling will reset all history calls [getHistory] first before toggling to retain a copy.
     *
     * @param isEnabled set this to true to turn on history recording and false to turn it off
     */
```